Assets/_Game/Scripts/UI/ActivationPopupUI.cs

```diff
@@ -0,0 +1,86 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class ActivationPopupUI : MonoBehaviour
+{
+    [SerializeField]
+    private Animator anim;
+    [SerializeField]
+    private PlayerInput playerInput;
+
+    private void OnEnable()
+    {
+        if(playerInput != null)
+        {
+            playerInput.onControlsChanged += PlayerInput_onControlsChanged;
+
+            HandleInput(playerInput);
+        }
+    }
+
+    private void OnDisable()
+    {
+        if (playerInput != null)
+        {
+            playerInput.onControlsChanged -= PlayerInput_onControlsChanged;
+        }
+    }
+
+    private void Awake()
+    {
+        anim = GetComponent<Animator>();
+        playerInput = FindObjectOfType<PlayerInput>();
+    }
+
+    private void PlayerInput_onControlsChanged(PlayerInput input)
+    {
+        HandleInput(input);
+    }
+
+    public void HandleInput(PlayerInput input)
+    {
+        if (input.currentControlScheme == "Keyboard&Mouse") { ChangeUIPrompt("KEYBOARD"); }
+        if (input.currentControlScheme == "Gamepad")
+        {
+            if (Gamepad.current.device.name.Contains("XInput")) { ChangeUIPrompt("XINPUT"); }
+            else if (Gamepad.current.device.name.Contains("DualShock")) { ChangeUIPrompt("DUALSHOCK"); }
+            else if (Gamepad.current.device.name.Contains("Switch")) { ChangeUIPrompt("SWITCH"); }
```

Comment:
Anytime you're doing string comparison, its a good idea to ToLower it beforehand just in case the string is not exactly the same case as you expect.

Also, do a local variable for Gamepad.current.device.name before the if statements so you don't keep querying potential lookups to the Gamepad.

Also, recommend changing the explicit strings into const or static readonly variables that are either declared in this class if its the only one using them or in a separate helper structure that can use them as static variables

Reply:
I changed to a toLower and cached Gamepad.current.device.name, and changed the strings to be private const strings